lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -75,20 +90,38 @@ module.exports = function (
       return $q.resolve(false)
     }
 
-    if (constraints.isDriveSizeRecommended(drive, selectionState.getImage())) {
-      return $q.resolve(true)
+    const isLargeDrive = constraints.isDriveSizeLarge(drive)
+    const isRecommendedDrive = constraints.isDriveSizeRecommended(drive, selectionState.getImage())
+
+    if (isLargeDrive || !isRecommendedDrive) {
+      const messageTuples = [
+        [
+          isLargeDrive,
+          messages.warning.largeDriveSize({
+            humanDriveSize: units.bytesToClosestUnit(drive.size)
+          })
+        ],
+        [
+          !isRecommendedDrive,
+          messages.warning.unrecommendedDriveSize({
+            drive,
+            image: selectionState.getImage()
+          })
+        ]
+      ]
+
+      const concatenatedMessage = _.reduce(messageTuples, (accumulatedMessage, messageTuple) => {
```

Comment:
Do we really need to concatenate the messages? Seems _extremely_ unlikely that a drive will be both 'large' and 'smaller than the recommended size' ;-)